### ISSUE TITLE
Linting errors for `_typeshed` on stub generated files

### ISSUE THREAD

**Issue by ramarivera** [PyCQA/pylint#7619]
### Question

I'm creating a type-hinting stub for some library I installed, using `stubgen -p <some_lib>`. 

The stub tool generated a bunch of classes, etc, and a single import at the top of the file:

```python
from _typeshed import Incomplete

class Maybe: ...
```

However, pylint is screaming about that import statement with `Unable to import '_typeshed'pylint(import-error)` (I'm using it on VS Code).

When running it on the command line, I get the following:

```
$ poetry run python -m pylint ./stubs/pymaybe/__init__.pyi 
************* Module __init__
stubs\pymaybe\__init__.pyi:5:0: E0401: Unable to import '_typeshed' (import-error)
```

So, I'm wondering if I'm doing something wrong and pylint should ignore that directory, if I should ignore the warning, or if it is a bug.

As per [typeshed docs](https://github.com/python/typeshed/tree/master/stdlib/_typeshed#usage) it seems that module doesn't exist (only at runtime), so maybe that's why?

### Documentation for future user

Not sure TBH, but maybe add a comment somewhere that the directory should be ignored?
My google search didn't bring anything relevant to `pylint + typeshed` tho

**Comment by clavedeluna** [comment]
I haven't personally used typeshed so MMMV but from spending time on other pylint issues, pylint (but specifically astroid) needs to "know" about types and classes. See issues with label `Needs Astroid Brain`.

Given these are runtime-only, adding `_typeshed`  to ignore configuration, something like:
```
# Files or directories to be skipped. They should be base names, not paths.
ignore=_typeshed
```

**Comment by Pierre-Sassoulas** [comment]
Imo it's due to .pyi file not being taken into account in pylint see (the currently pinned) https://github.com/PyCQA/pylint/issues/4987.
